Mathlib/Analysis/Convex/Cone/InnerDual.lean

```diff
@@ -195,7 +176,7 @@ theorem ConvexCone.hyperplane_separation_of_nonempty_of_isClosed_of_notMem (K :
     rwa [add_sub_cancel_right, real_inner_comm, ← neg_nonneg, neg_eq_neg_one_mul,
       ← real_inner_smul_right, neg_smul, one_smul, neg_sub] at hinner
   · -- as `K` is closed and non-empty, it is pointed
-    have hinner₀ := hinner 0 (K.pointed_of_nonempty_of_isClosed ne hc)
+    have hinner₀ := hinner 0 (.of_nonempty_of_isClosed ne hc : K.Pointed)
```

Comment:
Isn't there a way to write this without the explicit type ascription? If not, this is a strong sign that the old name was better.

Reply:
Do you consider this as better? If so, feel free to commit the suggestion
```suggestion
    have hinner₀ := hinner 0 (.of_nonempty_of_isClosed (C := K) ne hc)
```

Of course the issue is that `C.Pointed` is a somewhat silly definition, and all non-cone lemmas expect `0 ∈ C` instead